### PR TITLE
SV-1283 Various mods for Find an EPAO - standard versions updates

### PR DESCRIPTION
**Pull request by anilkdhiman6**

### CODE REVIEW

File: src/SFA.DAS.FindEpao.Web/Models/CourseEpaoViewModel.cs [hunk 1]
```diff
@@ -1,14 +1,79 @@
...
+        }
+
         public CourseListItemViewModel Course { set; get; }
         public EpaoDetailsViewModel Epao { get; set; }
         public int CourseEpaosCount { get; set; }
         public DateTime EffectiveFrom { get; set; }
         public IReadOnlyList<CourseListItemViewModel> AllCourses { get; set; }
+        public List<EpaoStandardsListItem> standardVersions { get; set; }
+        public List<ApprenticeShipTrainingCourses> apprenticeshipTrainingCourses {get; set;}
+        public List<ApprenticeShipTrainingCourses> allVersions { get; set; }
```
Comment (ben1stone) [inline src/SFA.DAS.FindEpao.Web/Models/CourseEpaoViewModel.cs:23]: Properties should start uppercase

**Review by ben1stone — CHANGES_REQUESTED** [review]

**Comment by ben1stone** [comment]
There's also new content for the start page

File: src/SFA.DAS.FindEpao.Web/Models/CourseEpaoViewModel.cs [hunk 2]
```diff
@@ -1,14 +1,79 @@
...
+                    if (allVersions.Any(x => x.LarsCode == standard.LarsCode))
+                    {
+                        foreach (var vers in allVersions)
+                        {
+                            if (vers.LarsCode == standard.LarsCode && standard.Status == "Live")
+                            {
+                                vers.Version += ", " + standard.Version;
+                            }
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Models/CourseEpaoViewModel.cs:50]: In the constructor - this is an empty list to start with

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 3]
```diff
@@ -24,25 +43,21 @@
...
-                }
+                <br />
+                <p class="govuk-!-font-weight-bold das-no-wrap">Standard Version Details
+                <table class="govuk-body das-no-wrap">
+                    <tr><td class="govuk-summary-list__value">Version</td><td class="govuk-summary-list__value">Effective From</td><td class="govuk-summary-list__value">Effective To</td></tr>
+                    @foreach (var versions in Model.apprenticeshipTrainingCourses)
+                    {
+                        @if (versions.LarsCode.ToString() == Model.Course.Id)
+                        {
+                            <tr><td class="govuk-summary-list__value">@versions.Version</td><td class="govuk-summary-list__value">@versions.EffectiveFrom.Value.ToString("dd MMM yyyy")</td><td class="govuk-summary-list__value">@versions.EffectiveTo</td></tr>
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml:54]: Extension method added for formatting - also will check with Victor regarding formatting/wrapping

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 4]
```diff
@@ -24,25 +43,21 @@
...
-                else
-                {
-                    <p class="govuk-body govuk-!-margin-bottom-1">
-                        @Model.Epao.Name is the only end-point assessment organisation for @Model.Course.Title <span class="das-no-wrap">(level @Model.Course.Level).</span>
-                    </p>
-                }
+                <br />
+                <p class="govuk-!-font-weight-bold das-no-wrap">Standard Version Details
+                <table class="govuk-body das-no-wrap">
+                    <tr><td class="govuk-summary-list__value">Version</td><td class="govuk-summary-list__value">Effective From</td><td class="govuk-summary-list__value">Effective To</td></tr>
```
[the remark this answers is not in the document]
Comment (jamesmking) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml:49]: Sorry to stick my nose in but I saw the PR, and noticed a few things needed changing. These table cells should be changed to `<th>` with an appropriate scope attribute. There are a mixture of CSS classes that shouldn't be used. `das-no-wrap` shouldn't be used on a table.

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 5]
```diff
@@ -24,25 +43,21 @@
...
-                           asp-route="@RouteNames.CourseEpaos">View the other end-point assessment organisations</a>
-                    </p>
-                }
-                else
-                {
-                    <p class="govuk-body govuk-!-margin-bottom-1">
-                        @Model.Epao.Name is the only end-point assessment organisation for @Model.Course.Title <span class="das-no-wrap">(level @Model.Course.Level).</span>
-                    </p>
-                }
+                <br />
```
Comment (jamesmking) [inline src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml:46]: This `<br />` shouldn't be needed

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 6]
```diff
@@ -122,24 +137,23 @@
...
-                            <li>@course.Title <span class="das-no-wrap">(level @course.Level)</span></li>
-                        }
-                        </ul>
-                    }
-                    </dd>
-                </div>
+                    
+                    <p class="govuk-!-font-weight-bold das-no-wrap">Apprenticeship training courses<br />
+
+                        <table class="govuk-body das-no-wrap">
```
Comment (jamesmking) [inline src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml:146]: Again, the CSS classes and markup look wrong to me. Is it a table or a summary list? If its a table, it needs table headings

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 7]
```diff
@@ -24,25 +43,21 @@
...
+                <br />
+                <p class="govuk-!-font-weight-bold das-no-wrap">Standard Version Details
+                <table class="govuk-body das-no-wrap">
+                    <tr><td class="govuk-summary-list__value">Version</td><td class="govuk-summary-list__value">Effective From</td><td class="govuk-summary-list__value">Effective To</td></tr>
+                    @foreach (var versions in Model.apprenticeshipTrainingCourses)
+                    {
+                        @if (versions.LarsCode.ToString() == Model.Course.Id)
+                        {
+                            <tr><td class="govuk-summary-list__value">@versions.Version</td><td class="govuk-summary-list__value">@versions.EffectiveFrom.Value.ToString("dd MMM yyyy")</td><td class="govuk-summary-list__value">@versions.EffectiveTo</td></tr>
+                        }
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml:55]: Updated with gov uk table classes - removed summary list classes

File: src/SFA.DAS.FindEpao.Web/Models/CourseEpaoViewModel.cs [hunk 8]
```diff
@@ -1,14 +1,79 @@
...
+            apprenticeshipTrainingCourses = new List<ApprenticeShipTrainingCourses>();
+            allVersions = new List<ApprenticeShipTrainingCourses>(); 
+        }
+
         public CourseListItemViewModel Course { set; get; }
         public EpaoDetailsViewModel Epao { get; set; }
         public int CourseEpaosCount { get; set; }
         public DateTime EffectiveFrom { get; set; }
         public IReadOnlyList<CourseListItemViewModel> AllCourses { get; set; }
+        public List<EpaoStandardsListItem> standardVersions { get; set; }
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Models/CourseEpaoViewModel.cs:21]: AllCourses to be added back in - in next check-in

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpaos.cshtml [hunk 9]
```diff
@@ -54,8 +54,13 @@
...
-                                    <dt class="das-definition-list__title govuk-!-margin-bottom-1">Address</dt><dd class="govuk-!-margin-bottom-1">@epao.Address</dd>
+
+                                    <dt class="das-definition-list__title govuk-!-margin-bottom-1">Locations covered</dt>
+                                    <dd class="govuk-!-margin-bottom-1">@if ((epao.Locations == null) || (epao.Locations == "")) { <text>&nbsp;</text>; } else { @epao.Locations; }</dd>
+                                    <dt class="das-definition-list__title govuk-!-margin-bottom-1">Address</dt>
+                                    <dd class="govuk-!-margin-bottom-1">@if ((epao.Address == null) || (epao.Address == "")) { <text>&nbsp;</text>; } else { @epao.Address; }</dd>
+                                    <dt class="das-definition-list__title govuk-!-margin-bottom-1">Versions</dt>
+                                    <dd class="govuk-!-margin-bottom-1">@if ((epao.Versions == null) || (epao.Versions == "")) { <text>&nbsp;</text>; } else { @epao.Versions; }</dd>
```
[the remark this answers is not in the document]
Comment (ben1stone) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpaos.cshtml]: This still doesn't match the content.

[the remark this answers is not in the document]
Comment (jamesmking) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpaos.cshtml]: `<text>` isn't valid html, use a `<span>` if you need to use a HTML element at all

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 10]
```diff
@@ -122,24 +137,23 @@
...
-                        @foreach (var course in Model.AllCourses)
-                        {
-                            <li>@course.Title <span class="das-no-wrap">(level @course.Level)</span></li>
-                        }
-                        </ul>
-                    }
-                    </dd>
-                </div>
+                    
+                    <p class="govuk-!-font-weight-bold das-no-wrap">Apprenticeship training courses<br />
```
[the remark this answers is not in the document]
Comment (ben1stone) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml]: Unresolved: this is still nested in p tag

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ben1stone) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml]: Unresolved: ToString("D") will output eg 'Monday, June 15, 2009'

[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml]: format changed to - dd MMMM yyyy

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpaos.cshtml (hunk 9 above)
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpaos.cshtml]: replaced <text> with <span>

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml (hunk 10 above)
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml]: This was resolved - there are no `<br>` tags or table inside `<p>` in the most recent commits

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 11]
```diff
@@ -52,14 +67,14 @@
...
             <dl class="govuk-summary-list govuk-!-margin-bottom-8">
                 @if (!string.IsNullOrEmpty(Model.Epao?.PrimaryContactName))
                 {
                     <div class="govuk-summary-list__row">
-                        <dt class="govuk-summary-list__key">Contact name</dt>
-                        <dd class="govuk-summary-list__value">@Model.Epao.PrimaryContactName</dd>
-                    </div>
+                    <dt class="govuk-summary-list__key">Contact name</dt>
+                    <dd class="govuk-summary-list__value">@Model.Epao.PrimaryContactName</dd>
+            </div>
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml]: tabs/spacing was updated in most recent commits

File: src/SFA.DAS.FindEpao.Web/Models/EpaoListItemViewModel.cs [hunk 12]
```diff
@@ -26,6 +28,8 @@ public class EpaoListItemViewModel : EpaoWithLocation
         public string Postcode { get; }
         public string Address => FormatAddress();
         public DateTime EffectiveFrom { get; set; }
+        public string[] StandardVersions {get; set;}
+        public string Versions { get; set; }
```
Comment (ben1stone) [inline src/SFA.DAS.FindEpao.Web/Models/EpaoListItemViewModel.cs:32]: The view model shouldn't need both Versions and StandardVersions? The Api returns the list of standards so they could be joined in the constructor or iterate over the list on the view. Happy to disucss

**Review by ben1stone — CHANGES_REQUESTED** [review]

**Review by jamesmking — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.FindEpao.Web/Models/EpaoListItemViewModel.cs (hunk 12 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Models/EpaoListItemViewModel.cs]: Removed Versions & added extension method to perform the array concatenation in the view

**Review by CodeWolf86 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 13]
```diff
@@ -1,48 +1,75 @@
...
-                }
-                else
-                {
-                    <p class="govuk-body govuk-!-margin-bottom-1">
-                        @Model.Epao.Name is the only end-point assessment organisation for @Model.Course.Title <span class="das-no-wrap">(level @Model.Course.Level).</span>
-                    </p>
-                }
+
+                <h2 class="govuk-heading-m">Standard version details</h2>
+                <table class="govuk-table" aria-describedby="StandardVersionsTable">
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml]: updated to title

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 14]
```diff
@@ -1,48 +1,75 @@
...
+                    <thead class="govuk-table__header" id="StandardVersionsHead" aria-label="StandardVersionDetails">
+                        <tr>
+                            <th id="headerVersion">Version</th>
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml]: added

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 15]
```diff
@@ -121,27 +148,29 @@
...
-                    }
-                    else
-                    {
-                        <ul class="govuk-list govuk-list--bullet">
-                        @foreach (var course in Model.AllCourses)
+
+                </dl>
+
+                <h2 class="govuk-heading-m">Apprenticeship training courses</h2>
+                <table class="govuk-table" aria-describedby="Apprenticeship training courses">
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml]: updated to title

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 16]
```diff
@@ -121,27 +148,29 @@
...
+                    <thead class="govuk-table__header" id="ApprenticeshipTrainingCoursesHead">
+                        <tr>
+                            <th id="Standard">Standard</th>
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml]: added

File: src/SFA.DAS.FindEpao.Domain/Courses/CourseEpao.cs [hunk 17]
```diff
@@ -13,5 +13,6 @@ public class CourseEpao
         public IReadOnlyList<EpaoDeliveryArea> EpaoDeliveryAreas { get; set; }
         public IReadOnlyList<DeliveryArea> DeliveryAreas { get; set; }
         public IReadOnlyList<CourseListItem> AllCourses { get; set; }
+        public List<EpaoStandardsListItem> standardVersions { get; set; }
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Domain/Courses/CourseEpao.cs]: updated

File: src/SFA.DAS.FindEpao.sln [hunk 18]
```diff
@@ -5,32 +5,32 @@ VisualStudioVersion = 16.0.30413.136
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SFA.DAS.FindEpao.Web", "SFA.DAS.FindEpao.Web\SFA.DAS.FindEpao.Web.csproj", "{A62A4E4D-537D-440A-B00F-42ACE7A3D21F}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SFA.DAS.FindEpao.Web.UnitTests", "SFA.DAS.FindEpao.Web.UnitTests\SFA.DAS.FindEpao.Web.UnitTests.csproj", "{0FF48677-BCDA-4A0E-A63B-6E81609B470D}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SFA.DAS.FindEpao.Web.UnitTests", "SFA.DAS.FindEpao.Web.UnitTests\SFA.DAS.FindEpao.Web.UnitTests.csproj", "{0FF48677-BCDA-4A0E-A63B-6E81609B470D}"
```
Comment (CodeWolf86) [inline src/SFA.DAS.FindEpao.sln:8]: How did all the Project Guids change in the sln file?

File: src/SFA.DAS.FindEpao.Application.UnitTests/Courses/Queries/GetCourseEpao/WhenHandlingGetCourseEpaoQuery.cs [hunk 19]
```diff
@@ -61,6 +61,7 @@ public class WhenHandlingGetCourseEpaoQuery
             result.DeliveryAreas.Should().BeEquivalentTo(courseEpao.DeliveryAreas);
```
Comment (CodeWolf86) [inline src/SFA.DAS.FindEpao.Application.UnitTests/Courses/Queries/GetCourseEpao/WhenHandlingGetCourseEpaoQuery.cs:61]: Does `StandardVersions` need to also be asserted?

File: src/SFA.DAS.FindEpao.Domain.UnitTests/Courses/WhenConstructingACourseListItem.cs [hunk 20]
```diff
@@ -58,7 +59,7 @@ public void Then_The_Empty_Constructor_Sets_The_Title_To_Unknown()
             var expectedString = "Unknown";
 
             //Act
-            var course = new CourseListItem(null, null, 0, false);
+            var course = new CourseListItem(null, null, 0, false, new string[] { "1.0" });
```
Comment (CodeWolf86) [inline src/SFA.DAS.FindEpao.Domain.UnitTests/Courses/WhenConstructingACourseListItem.cs:62]: Do you need a test for when the expected versions are null / empty like the other permutations?

File: src/SFA.DAS.FindEpao.Web.UnitTests/Controllers/CoursesControllerTests/WhenGettingCourseEpao.cs [hunk 21]
```diff
@@ -88,6 +88,7 @@ public class WhenGettingCourseEpao
             model!.EffectiveFrom.Should().Be(mediatorResult.EffectiveFrom);
             model!.AllCourses.Should().BeEquivalentTo(
                 mediatorResult.AllCourses.Select(item => (CourseListItemViewModel)item));
+
```
Comment (CodeWolf86) [inline src/SFA.DAS.FindEpao.Web.UnitTests/Controllers/CoursesControllerTests/WhenGettingCourseEpao.cs:91]: Do you need a Standard Version assert here?

File: src/SFA.DAS.FindEpao.Web/Extensions/DateFormatExtensions.cs [hunk 22]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace SFA.DAS.FindEpao.Web.Extensions
+{
+    public static class DateFormatExtensions
+    {
+        public static string LongDateFormat(this DateTime? sourceDateTime)
```
Comment (CodeWolf86) [inline src/SFA.DAS.FindEpao.Web/Extensions/DateFormatExtensions.cs:7]: Do you have any tests around the new format extensions?

File: src/SFA.DAS.FindEpao.Web/Extensions/StandardVersionsFormatExtensions.cs [hunk 23]
```diff
@@ -0,0 +1,11 @@
+﻿namespace SFA.DAS.FindEpao.Web.Extensions
+{
+    public static class StandardVersionsFormatExtensions
+    {
+        public static string VersionsArrayToString(string[] standardVersions)
+        {            
+            return standardVersions == null ? string.Empty : string.Join(", ", standardVersions);
```
Comment (CodeWolf86) [inline src/SFA.DAS.FindEpao.Web/Extensions/StandardVersionsFormatExtensions.cs:7]: Do you have any tests around the new format extensions?

File: src/SFA.DAS.FindEpao.sln [hunk 24]
```diff
@@ -5,32 +5,32 @@ VisualStudioVersion = 16.0.30413.136
...
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SFA.DAS.FindEpao.Web.AcceptanceTests", "SFA.DAS.FindEpao.Web.AcceptanceTests\SFA.DAS.FindEpao.Web.AcceptanceTests.csproj", "{E6170F8C-D5C4-4831-8F32-8CA08936A727}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SFA.DAS.FindEpao.Web.AcceptanceTests", "SFA.DAS.FindEpao.Web.AcceptanceTests\SFA.DAS.FindEpao.Web.AcceptanceTests.csproj", "{E6170F8C-D5C4-4831-8F32-8CA08936A727}"
```
Comment (CodeWolf86) [inline src/SFA.DAS.FindEpao.sln:10]: Can we back out this change? shouldn't be any changes to the sln file for the work done.

**Review by CodeWolf86 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.FindEpao.sln (hunk 24 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.sln:10]: Backed-out

File: src/SFA.DAS.FindEpao.sln (hunk 18 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.sln:8]: Backed-out

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml [hunk 25]
```diff
@@ -1,48 +1,75 @@
...
-                else
-                {
-                    <p class="govuk-body govuk-!-margin-bottom-1">
-                        @Model.Epao.Name is the only end-point assessment organisation for @Model.Course.Title <span class="das-no-wrap">(level @Model.Course.Level).</span>
-                    </p>
-                }
+
+                <h2 class="govuk-heading-m" id="StandVerDetails">Standard version details</h2>
+                <table class="govuk-table" aria-describedby="StandVerDetails">
+                    <thead class="govuk-table__header" id="StandardVersionsHead" aria-label="StandardVersionDetails">
```
Comment (jamesmking) [inline src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml:50]: Can we remove the `aria-label="StandardVersionDetails"`? It doesn't do anything, and will be read out by a screenreader

**Review by jamesmking — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.FindEpao.Application.UnitTests/Courses/Queries/GetCourseEpao/WhenHandlingGetCourseEpaoQuery.cs (hunk 19 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Application.UnitTests/Courses/Queries/GetCourseEpao/WhenHandlingGetCourseEpaoQuery.cs:61]: updated

File: src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml (hunk 25 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Views/Courses/CourseEpao.cshtml:50]: removed `aria-label="StandardVersionDetails"`

File: src/SFA.DAS.FindEpao.Web.UnitTests/Controllers/CoursesControllerTests/WhenGettingCourseEpao.cs (hunk 21 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web.UnitTests/Controllers/CoursesControllerTests/WhenGettingCourseEpao.cs:91]: updated

File: src/SFA.DAS.FindEpao.Domain.UnitTests/Courses/WhenConstructingACourseListItem.cs (hunk 20 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Domain.UnitTests/Courses/WhenConstructingACourseListItem.cs:62]: test added

File: src/SFA.DAS.FindEpao.Web/Extensions/StandardVersionsFormatExtensions.cs (hunk 23 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Extensions/StandardVersionsFormatExtensions.cs:7]: added

File: src/SFA.DAS.FindEpao.Web/Extensions/DateFormatExtensions.cs (hunk 22 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Extensions/DateFormatExtensions.cs:7]: tests added

**Review by jamesmking — COMMENTED** [review]
Looks ok from a FE point of view

**Review by jamesmking — APPROVED** [review]

File: src/SFA.DAS.FindEpao.Web/Extensions/DateFormatExtensions.cs (hunk 22 above)
Comment (ben1stone) [reply src/SFA.DAS.FindEpao.Web/Extensions/DateFormatExtensions.cs:7]: Could we pull apart the tests for the Date and String version extensions into their own files. I think the date test should also define the expected outcome in the test case, because at the moment it is repeating the logic it is trying to test.

**Review by CodeWolf86 — APPROVED** [review]

File: src/SFA.DAS.FindEpao.Web/Extensions/DateFormatExtensions.cs (hunk 22 above)
Comment (anilkdhiman6) [reply src/SFA.DAS.FindEpao.Web/Extensions/DateFormatExtensions.cs:7]: split files for extension methods (Date and String). Also added explicit check for date

**Review by ben1stone — APPROVED** [review]
